keyboards/splitkb/zima/zima.c

```diff
@@ -0,0 +1,112 @@
+/* Copyright 2019 Thomas Baart
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "zima.h"
+#include <stdio.h>
+
+#ifdef HAPTIC_ENABLE
+#    include "haptic.h"
+extern haptic_config_t haptic_config;
+#endif
+
+#ifdef OLED_DRIVER_ENABLE
+static bool is_asleep = false;
+static uint32_t oled_timer;
+
+void suspend_power_down_kb(void) {
+    is_asleep = true;
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    is_asleep = false;
+    suspend_wakeup_init_user();
+}
+
+__attribute__((weak)) oled_rotation_t oled_init_user(oled_rotation_t rotation) { return OLED_ROTATION_180; }
+
+__attribute__((weak)) void oled_task_user(void) {
+    if (is_asleep) {
+        oled_off();
+        return;
+    }
+
+    if (timer_elapsed32(oled_timer) < 30000) {
+        oled_on();
+        oled_scroll_off();
+        oled_write_P(PSTR("SplitKB's Zima"), false);
+        char layer[2] = {0};
+        snprintf(layer, sizeof(layer), "%d", get_highest_layer(layer_state));
+        oled_write_P(PSTR("   L:"), false);
+        oled_write_ln(layer, false);
+        oled_write_ln_P(PSTR("--------------"), false);
+        if (rgblight_is_enabled()) {
+            oled_write_P(PSTR("HSV: "), false);
+            char rgbs[14];
+            snprintf(rgbs, sizeof(rgbs), "%3d, %3d, %3d", rgblight_get_hue(), rgblight_get_sat(), rgblight_get_val());
+            oled_write_ln(rgbs, false);
+        } else {
+            oled_write_ln_P(PSTR("RGB LIGHT DISABLED"), false);
+        }
+#    ifdef AUDIO_ENABLE
+        oled_write_P(PSTR("Audio:"), false);
```

Comment:
As strings directly take up MCU space afaik, I suggest you take the approach to limit Strings as far as possible
```C
if(is_audio_on())
  oled_write_P(PSTR("audio");
```
(also it would reduce screen clutter a bit ;)
Same applies to clicky

If you stay with the approach, you should add a space: `"off" - > " off"`

Reply:
It does, but it's a byte per character. As for changing it, I think it's worth the couple extra bytes to make it look better, here.

Reply:
The default keymap is primarily to be used with the configurator. Enthusiasts who need the space will redefine methods and strip the keymap to their own minimum. As such, optimizing per character isn't necessary in this particular instance.